src/IBM.Cloud.SDK.Core/Service/IBMService.cs

```diff
@@ -60,14 +60,14 @@ protected string Endpoint
 
         private IAuthenticator authenticator;
 
-        protected IBMService(string serviceName, string url, IClient httpClient)
+        protected IBMService(string serviceName, string serviceUrl, IClient httpClient)
         {
             ServiceName = serviceName;
             Client = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
             authenticator = new NoAuthAuthenticator();
 
             if (!string.IsNullOrEmpty(Endpoint))
-                Endpoint = url;
+                Endpoint = serviceUrl;
```

Comment:
Is this the correct conditional check here???
I read this as... "If the 'Endpoint' value currently set on this IBMService instance is not null or empty, then assign the serviceUrl value to it", which doesn't make sense.

In fact, I'm not quite sure we want to pass in the serviceUrl parameter to the constructor anyway.
Why not let the IBMService instance be constructed with a null Endpoint/ServiceURL value, then let the
generator decide if it wants to set the default service url by generating the code to do that.
So, I guess I'm suggesting we simply omit the "serviceUrl" parameter to this constructor altogether.

Reply:
This PR is _only_ for refactoring SetEndpoint to SetServiceUrl. I'm working the core changes for url for another PR.